Sources/Atomic.swift

```diff
@@ -8,56 +8,109 @@
 
 import Foundation
 
-final class PosixThreadMutex: NSLocking {
-	private var mutex = pthread_mutex_t()
+internal struct _PosixThreadMutex {
+	private let mutex: UnsafeMutablePointer<pthread_mutex_t>
 
 	init() {
-		let result = pthread_mutex_init(&mutex, nil)
+		mutex = UnsafeMutablePointer<pthread_mutex_t>.allocate(capacity: 1)
+		mutex.initialize(to: pthread_mutex_t())
+		let result = pthread_mutex_init(mutex, nil)
 		precondition(result == 0, "Failed to initialize mutex with error \(result).")
 	}
 
-	deinit {
-		let result = pthread_mutex_destroy(&mutex)
+	func deinitialize() {
+		let result = pthread_mutex_destroy(mutex)
 		precondition(result == 0, "Failed to destroy mutex with error \(result).")
+		mutex.deinitialize()
+		mutex.deallocate(capacity: 1)
+	}
+
+	@inline(__always)
+	func lock() {
+		let result = pthread_mutex_lock(mutex)
+		if result != 0 {
+			fatalError("Failed to lock \(self) with error \(result).")
+		}
+	}
+
+	@inline(__always)
+	func unlock() {
+		let result = pthread_mutex_unlock(mutex)
+		if result != 0 {
+			fatalError("Failed to unlock \(self) with error \(result).")
+		}
+	}
+}
+
+internal final class PosixThreadMutex: NSLocking {
```

Comment:
It doesn't look like this is used anywhere? Let's remove it and then remove the underscore from the version we _are_ using.